feat: 🎉 `blurAuthor` plugin

TobyBridle opened the pull request:
This adds the ability to set author(s) which have their attachments automatically hidden, similar to the existing `blurNsfw` plugin.

In addition to setting it within the Plugin configuration page, it adds an action to the context menu to add/remove a given user to the blocklist.

Photos:
- Hidden by default
- Shown when the IMAGE (not the message) is hovered

Videos:
- Hidden by default
- Shown when playing, hovered or in fullscreen

*I understand that there may not be an open request for this. I was interested in some plugin development and wanted to do it for personal uses*

TobyBridle committed:
feat: `blurAuthor` plugin

This adds the ability to set author(s) which have their attachments
automatically hidden, similar to the existing `blurNsfw` plugin.

In addition to setting it within the Plugin configuration page, it adds
an action to the context menu to add/remove a given user to the
blocklist.

Photos:
- Hidden by default
- Shown when the IMAGE (not the message) is hovered

Videos:
- Hidden by default
- Shown when playing, hovered or in fullscreen

sadan4 commented on src/plugins/blurAuthor/index.ts:
```diff
@@ -0,0 +1,108 @@
+    },
+
+    start() {
+        style = document.createElement("style");
```
import a css file instead of using dom mainp

sadan4 replied:
nevermind, im stupid

didnt notice the template string :sob:

TobyBridle committed:
fix(blurAuthor): 🐛 remove unnecessary truthy check

`ids` is always a string, removes the unnecessary (and ugly :( ) check
for truthy-ness

Vendicated commented:
The BlurNSFW plugin has awful code and nothing new should be based on it. We also don't allow plugins that are just slight modifications of others. It would make much more sense to just add this functionality to the original plugin. Could just add an option to mark users as NSFW

TobyBridle commented:
> The BlurNSFW plugin has awful code and nothing new should be based on it. We also don't allow plugins that are just slight modifications of others. It would make much more sense to just add this functionality to the original plugin. Could just add an option to mark users as NSFW

If that's welcome, I'm perfectly fine with doing that
